backends/p4tools/common/lib/logging.h

```diff
@@ -9,11 +9,41 @@
 
 namespace P4Tools {
 
-/// Helper functions that prints strings associated with basic test generation information., for
-/// example the covered nodes or tests number.
+/// Helper function for @printFeature
+inline std::string logHelper(boost::format &f) { return f.str(); }
+
+/// Helper function for @printFeature
+template <class T, class... Args>
+std::string logHelper(boost::format &f, T &&t, Args &&...args) {
+    return logHelper(f % std::forward<T>(t), std::forward<Args>(args)...);
+}
+
+/// A helper function that allows us to configure logging for a particular feature. This code is
+/// taken from
+// https://stackoverflow.com/a/25859856
+template <typename... Arguments>
+void printFeature(const std::string &label, int level, const std::string &fmt,
+                  Arguments &&...args) {
+    boost::format f(fmt);
+
+    auto result = logHelper(f, std::forward<Arguments>(args)...);
+
+    LOG_FEATURE(label.c_str(), level, result);
```

Comment:
It would be better to ask if the feature is enabled at given level so that the formatting is not done if the log is disabled. That would require pulling the condition out of `LOG_FEATURE` to something like `LOGGING_FEATURE`.

Reply:
Yeah, was thinking of doing that, too. I will see whether I can make this change without breaking existing calls.